server/service/team_policies.go

```diff
@@ -368,7 +368,8 @@ func (svc *Service) modifyPolicy(ctx context.Context, teamID *uint, id uint, p f
 		})
 	}
 
-	var shouldRemoveAll bool
+	var removeAllMemberships bool
+	var removeStats bool
```

Comment:
Is there a scenario where we want to remove all membership (`policy_membership`) but not stats (`policy_stats`)? 
Or a scenario where we want to remove `policy_stats` but not `policy_membership`?

AFAICS there should be one `bool` and it should remove both both.

Reply:
When updating platform, we only remove items from `policy_membership` which do not match the new platform field.

Reply:
We do clear the stats when updating platform. That's what this PR does.

Reply:
There is no scenario when we change `policy_membership` but not stats.

Reply:
Ah gotcha, I was confused, will be re-reviewing shortly.